lib/go/test/flow_idtable_staking_test.go

```diff
@@ -2303,3 +2303,246 @@ func TestIDTableStaking(t *testing.T) {
+	require.NoError(t, err)
+	signAndSubmit(
+		t, b, tx,
+		[]flow.Address{b.ServiceKey().Address, idTableAddress},
+		[]crypto.Signer{b.ServiceKey().Signer(), IDTableSigner},
+		false,
+	)
+
+	t.Run("Should be able to set multiple nodes as inactive so their rewards will be withheld", func(t *testing.T) {
+
+		tx := createTxWithTemplateAndAuthorizer(b, templates.GenerateSetNonOperationalScript(env), idTableAddress)
+		err := tx.AddArgument(cadence.NewArray([]cadence.Value{CadenceString(ids[0]), CadenceString(ids[2])}))
+		require.NoError(t, err)
+		signAndSubmit(
+			t, b, tx,
+			[]flow.Address{b.ServiceKey().Address, idTableAddress},
+			[]crypto.Signer{b.ServiceKey().Signer(), IDTableSigner},
+			false,
+		)
+
+		result := executeScriptAndCheck(t, b, templates.GenerateGetNonOperationalListScript(env), nil)
+
+		idArray := result.(cadence.Array).Values
+		assert.Equal(t, len(idArray), 2)
+		assert.Equal(t, idArray[0], CadenceString(ids[0]))
+		assert.Equal(t, idArray[1], CadenceString(ids[2]))
+	})
+
+	t.Run("Should pay rewards, but rewards from bad nodes and delegators are re-distributed", func(t *testing.T) {
+		// Pay Rewards
+		tx := createTxWithTemplateAndAuthorizer(b, templates.GeneratePayRewardsScript(env), idTableAddress)
+		signAndSubmit(
+			t, b, tx,
+			[]flow.Address{b.ServiceKey().Address, idTableAddress},
+			[]crypto.Signer{b.ServiceKey().Signer(), IDTableSigner},
+			false,
+		)
+
+		result := executeScriptAndCheck(t, b, templates.GenerateGetNonOperationalListScript(env), nil)
+
+		idArray := result.(cadence.Array).Values
+		assert.Equal(t, len(idArray), 0)
+
+		// Check Rewards balances of nodes and delegators who were withheld
+		// to make sure they got zero
+
+		// Node 0
+		result = executeScriptAndCheck(t, b, templates.GenerateGetRewardBalanceScript(env), [][]byte{jsoncdc.MustEncode(cadence.String(ids[0]))})
+		assertEqual(t, CadenceUFix64("0.0"), result)
+
+		// Node 2
+		result = executeScriptAndCheck(t, b, templates.GenerateGetRewardBalanceScript(env), [][]byte{jsoncdc.MustEncode(cadence.String(ids[2]))})
+		assertEqual(t, CadenceUFix64("0.0"), result)
+
+		// Delegators from Node 0
+		for i := 1; i < 6; i++ {
+			result = executeScriptAndCheck(t, b, templates.GenerateGetDelegatorRewardsScript(env), [][]byte{jsoncdc.MustEncode(cadence.String(ids[0])), jsoncdc.MustEncode(cadence.UInt32(i))})
+			assertEqual(t, CadenceUFix64("0.0"), result)
+		}
+
+		// Check rewards balances of the other nodes and delegators to make sure
+		// the extra rewards were distributed to them
+
+		// Calculate Rewards
+		// All the nodes staked 500k FLOW and all the delegators staked 10k FLOW
+		// so the calculations are slightly easier
+		// Node 0 has 5 delegators, node 1 has 5 delegators,
+		// and the other nodes have no delegators
+		// Tested witholding rewards from one of the nodes with delegators
+		// and one of the non-delegated-to nodes
+
+		var delegatorCommitment interpreter.UFix64Value = 1000000000000
```

Comment:
For this and the following `interpreter.UFix64Value`-typed values: please add a comment noting what the value is or maybe a helper function which converts a more readable numeric representation to the `interpreter.UFix64Value` type